etl-routes.js

```diff
@@ -692,38 +695,50 @@ module.exports = function () {
                     },
                     handler: function (request, reply) {
                         let EIDLabReminderService = require('./service/eid/eid-lab-reminder.service');
-                        EIDLabReminderService.pendingEIDReminders(request.params, config.hivLabSystem)
-                            .then((eidReminders) => {
-                                let combineRequestParams = Object.assign({}, request.query, request.params);
-                                combineRequestParams.limitParam = 1;
-                                let reportParams = etlHelpers.getReportParams('clinical-reminder-report', ['referenceDate', 'patientUuid', 'offSetParam', 'limitParam'], combineRequestParams);
-
-                                let report = new BaseMysqlReport('clinicalReminderReport', reportParams.requestParams);
-                                report.generateReport().then((results) => {
-                                    try {
-                                        if (results.results.results.length > 0) {
-                                            let processedResults = patientReminderService.generateReminders(results.results.results, eidReminders);
-                                            results.result = processedResults;
-                                        } else {
-                                            results.result = {
-                                                person_uuid: combineRequestParams.person_uuid,
-                                                reminders: []
-                                            };
-                                        }
-                                        reply(results);
-                                    } catch (error) {
-                                        console.log('Error generating reminders', error);
-                                        reply(Boom.badImplementation('An internal error occurred'));
-                                    }
+                        let combineRequestParams = Object.assign({}, request.query, request.params);
+                        syncPreproc.processPendingLabResultRequest(request.params.patientUuid)
+                        .then((cachedResults)=> {
+                             if(cachedResults.length > 0){
+                                EIDLabReminderService.generateRemindersReport(combineRequestParams,cachedResults)
+                                .then((report) => {
```

Comment:
Adding the 'cachedResults' to the signature of generateRemindersReport will introduce maintenance challenges later on. We need to take the variable closer to where it's used to avoid complicating the signature of root functions. Somewhere inside the generateRemindersReport, there should be a call to get pending lab order, in that call, you can do the caching inside that function.